### PR TITLE
build(gha): Replace eslint action with workflow

### PR DESCRIPTION
**Pull request by billyvg**
This replaces the external eslint action with steps defined in the workflow. This also fixes an issue where the eslint action was not erroring when there were only formatting errors that were auto fixed.

### CODE REVIEW

**Comment by billyvg** [comment]
You can see it working here: https://github.com/billyvg/sentry/pull/49

File: .github/workflows/js-build-and-lint.yml [hunk 1]
```diff
@@ -55,14 +59,34 @@ jobs:
...
+        run: |
+          # Run relax config on main branch (and stricter config for changed files)
+          yarn lint -c .eslintrc.relax.js
+          yarn lint:css
+
+      # Otherwise if it's not main branch, only lint modified files
+      - name: eslint (changed files only)
+        if: github.ref != 'refs/heads/master' && steps.changes.outputs.frontend == 'true'
+        run: |
+          yarn eslint --fix ${{ steps.changes.outputs.frontend_modified_lintable_files }}
```
Comment (joshuarli) [inline .github/workflows/js-build-and-lint.yml:74]: This is the only occurence of `frontend_modified_lintable_files`. Where does it come from?

File: .github/workflows/js-build-and-lint.yml [hunk 2]
```diff
@@ -55,14 +59,34 @@ jobs:
...
+
+      # Otherwise if it's not main branch, only lint modified files
+      - name: Commit any eslint fixed files
+        if: github.ref != 'refs/heads/master' && steps.changes.outputs.frontend == 'true'
+        continue-on-error: true
+        run: |
+          git config --global user.name "github-actions[bot]"
+          git config --global user.email "github-actions@sentry.io"
+          git add -A
+          git commit -m "style(): Auto eslint fix ${GITHUB_SHA}" || exit 0
```
Comment (joshuarli) [inline .github/workflows/js-build-and-lint.yml:84]:
```suggestion
          git commit -m "chore: Automatic eslint fix ${GITHUB_SHA}" || exit 0
```

File: .github/workflows/js-build-and-lint.yml [hunk 3]
```diff
@@ -55,14 +59,34 @@ jobs:
...
+          git push origin "HEAD:${{ github.event.pull_request.head.ref }}"
```
Comment (joshuarli) [inline .github/workflows/js-build-and-lint.yml:85]: Is there a need to be so explicit here?

File: .github/workflows/js-build-and-lint.yml (hunk 1 above)
Comment (billyvg) [reply .github/workflows/js-build-and-lint.yml:74]: It comes from the `paths-filter` action, it provides an output of the files: https://github.com/dorny/paths-filter#custom-processing-of-changed-files

Comment (joshuarli) [reply .github/workflows/js-build-and-lint.yml:74]: Oh, I see: `${FILTER_NAME}_files`

File: .github/workflows/js-build-and-lint.yml (hunk 3 above)
Comment (billyvg) [reply .github/workflows/js-build-and-lint.yml]: There was but I changed the checkout to checkout b ranch instead of a commit so we can remove this.

**Review by joshuarli — APPROVED** [review]

**Comment by kamilogorek** [comment]
It's so beautiful. We need this in `sentry-javascript` so bad! <3

**Comment by billyvg** [comment]
@kamilogorek lol, it works ok when it's only a small # of files, but gets buggy when there are lots.

**Comment by kamilogorek** [comment]
Thats enough, as it allows you to do quick fixes through GitHub UI without worrying about linters though.

**Pull request merged**